platforms/browser/www/phonegap/plugin/facebookConnectPlugin/facebookConnectPlugin.js

```diff
@@ -161,16 +161,31 @@ cordova.define("com.phonegap.plugins.facebookconnect.FacebookConnectPlugin", fun
         
         // Bake in the JS SDK
         (function () {
+            // Retrieve the root element to append the script tags to. Use the body if no fb-root was found
+            var root = document.getElementById('fb-root') || document.getElementsByTagName('body')[0];
+
             if (!window.FB) {
                 console.log("launching FB SDK");
                 var e = document.createElement('script');
-                e.src = document.location.protocol + '//connect.facebook.net/en_US/sdk.js';
                 e.async = true;
-                document.getElementById('fb-root').appendChild(e);
+                e.src = document.location.protocol + '//connect.facebook.net/en_US/sdk.js';
+                e.onload = e.onreadystatechange = e.onerror = loadLocalSDK;
+
+                root.appendChild(e);
+            }
+
+            /**
+             * If something happens when loading the remote script, then check if
+             * window.FB exists, if not, load the local script.
+             */
+            function loadLocalSDK() {
                 if (!window.FB) {
-                    // Probably not on server, use the sample sdk
+                    var e = document.createElement('script');
+                    e.async = true;
                     e.src = 'phonegap/plugin/facebookConnectPlugin/fbsdk.js';
```

Comment:
This I think is an error on my behalf that has now caught my attention. In fact, this file does not exist.
I also think that a fallback is not a requirement on the browser platform for the following reasons:
- there are very few cases this would file would not be loaded eg.;
    - Facebook stops serving the file
    - timeout (in which case the whole site would timeout)
    - bad proxy
All of the above would mean window.FB is undefined. However this error would be caught by the try catch when calling the APIs.

If the above is true then `loadLocalSDK ` is no longer required as is tracking `e.onerror`.